third_party/move/move-vm/runtime/src/move_vm.rs

```diff
@@ -107,6 +111,16 @@ impl MoveVM {
         self.runtime.loader().get_and_clear_module_cache_hits()
     }
 
+    pub fn get_ref(&self) -> MoveVMRef {
```

Comment:
not sure I understand why we need this change? we can get &MoveVM from Arc<MoveVM> directly by as_ref()?

Reply:
This relates to my question on Rati's PR why we have VMRuntime and MoveVM as a wrapper.

We also want to be able to refer to MoveVM from Session, but Session stores VMRuntime so there is no way we can get a &MoveVM from a session. For getting metadata all we need is a reference to the Loader / VMRuntime. But those are crate local so we need a wrapper around them

Reply:
Yeah, this is also my biggest concern. `MoveVMRef` makes the the API a lot more complicated, so instead of doing this, we should simply expose the specific functionality (metadata lookup in this case) via both the `MoveVM` & the `Session` instead. 

Here's a working patch: https://github.com/aptos-labs/aptos-core/compare/gerben-fix-resource-groups...vgao1996:metadata-cleanup?expand=1

I know that this require us to store a `&MoveVM` in `SessionExt`, but it's just an extra reference and will not slow us down.

Reply:
Made session carry movevm, as an added benefit we can remove the useless indirection of new_session through movevm to runtime